lib/ex_aws/s3/client.ex

```diff
@@ -411,6 +416,16 @@ defmodule ExAws.S3.Client do
   defcallback list_parts(bucket :: binary, object :: binary, upload_id :: binary, opts :: Keyword.t) :: ExAws.Request.response_t
 
   @doc """
+  Generates a pre-signed URL for this object.
+
+  Raises an ArgumentError if `:expires_in` exceeds one week.
+
+  When option param :virtual_host is `true`, the {#bucket} name will be used as
+  the hostname. This will cause the returned URL to be 'http' and not 'https'.
+  """
+  defcallback presigned_url(http_method :: atom, bucket :: binary, object :: binary, opts :: presigned_url_opts) :: binary
```

Comment:
Related to the comment on the implementation, I believe the return specs here should be `{:ok, binary} | {:error, binary}`